src/nupic/types/Sdr.cpp

```diff
@@ -0,0 +1,710 @@
+/* ---------------------------------------------------------------------
+ * Copyright (C) 2018-2019, David McDougall.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * ----------------------------------------------------------------------
+ */
+
+/** @file
+ * Definitions for SparseDistributedRepresentation class
+ */
+
+#include "nupic/types/Sdr.hpp"
+
+#include <numeric>
+#include <algorithm> // std::sort
+
+using namespace std;
+
+namespace nupic {
+// SDR implementation in .cpp, refer to .hpp for detailed comments
+
+/**
+ * SparseDistributedRepresentation class
```

Comment:
Can we have the documentation in the header files only?  This is a lot of duplicate text...

Reply:
you were saying that you'd miss doc with the sources...but I agree it's duplicit and we'd get derailed in the texts. So remove and have it in hpp only?

Reply:
> have [docs] in hpp only?

Yes please.